kinesumer_test.go

```diff
@@ -308,3 +315,370 @@ func TestShardsRebalancing(t *testing.T) {
+
+	expectedSeqNum := "12345"
+	for _, client := range clients {
+		shardIDs := client.shards["events"].ids()
+		for _, shardID := range shardIDs {
+			env.client1.MarkRecord(&Record{
+				Stream:  "events",
+				ShardID: shardID,
+				Record: &kinesis.Record{
+					SequenceNumber: &expectedSeqNum,
+				},
+			})
+		}
+	}
+
+	for _, client := range clients {
+		client.Commit()
+	}
+
+	for _, client := range clients {
+		shardIDs := client.shards["events"].ids()
+		checkpoints, _ := client.stateStore.ListCheckPoints(context.Background(), "events", shardIDs)
+		for _, checkpoint := range checkpoints {
+			assert.EqualValues(t, expectedSeqNum, checkpoint, "sequence number should be equal")
+		}
+	}
+}
+
+func TestKinesumer_commitCheckPointPerStreamWorksProperly(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	input := []*ShardCheckPoint{
+		{
+			Stream:         "foobar",
+			ShardID:        "shardId-0",
+			SequenceNumber: "0",
+		},
+	}
+
+	mockStateStore := NewMockStateStore(ctrl)
+	mockStateStore.EXPECT().
+		UpdateCheckPoints(gomock.Any(), input).
+		Times(1).
+		Return(nil)
+
+	offsets := map[string]*sync.Map{}
+	offsets["foobar"] = &sync.Map{}
+	offsets["foobar"].Store("shardId-0", "0")
+	offsets["foobar"].Store("shardId-1", "1")
+	kinesumer := &Kinesumer{
+		offsets:       offsets,
+		commitTimeout: 2 * time.Second,
+		stateStore:    mockStateStore,
+	}
+
+	kinesumer.commitCheckPointsPerStream("foobar", input)
+
+	select {
+	case err := <-kinesumer.Errors():
+		assert.NoError(t, err, "there should be no error")
+	default:
+	}
+}
+
+func TestKinesumer_commitCheckPointPerStreamFails(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	testCases := []struct {
+		name         string
+		newKinesumer func() *Kinesumer
+		input        struct {
+			stream      string
+			checkpoints []*ShardCheckPoint
+		}
+		wantErrMsg string
+	}{
+		{
+			name: "when state store fails to update checkpoints",
+			newKinesumer: func() *Kinesumer {
+				mockStateStore := NewMockStateStore(ctrl)
+				mockStateStore.EXPECT().
+					UpdateCheckPoints(gomock.Any(), gomock.Any()).
+					Times(1).
+					Return(errors.New("mock error"))
+				return &Kinesumer{
+					errors:     make(chan error, 1),
+					stateStore: mockStateStore,
+				}
+			},
+			input: struct {
+				stream      string
+				checkpoints []*ShardCheckPoint
+			}{
+				stream: "foobar",
+				checkpoints: []*ShardCheckPoint{
+					{
+						Stream:         "foobar",
+						ShardID:        "shardId-000",
+						SequenceNumber: "0",
+					},
+				},
+			},
+			wantErrMsg: "failed to commit on stream: foobar: mock error",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			kinesumer := tc.newKinesumer()
+			kinesumer.commitCheckPointsPerStream(tc.input.stream, tc.input.checkpoints)
+			result := <-kinesumer.errors
+			assert.EqualError(t, result, tc.wantErrMsg, "there should be an expected error")
+		})
+	}
+}
+
+func TestKinesumer_cleanupOffsetsWorksFine(t *testing.T) {
```

Comment:
Some tests ends with `WorksProperly`, but some tests ends with `WorksFine`. Please unify the convention.

Reply:
[b1a17fd](https://github.com/daangn/kinesumer/pull/12/commits/b1a17fd34a43b65666976ae8f951f03cf1a45e8c)